src/item.h

```diff
@@ -2658,6 +2658,7 @@ class item : public visitable
         char invlet = 0;      // Inventory letter
         bool active = false; // If true, it has active effects to be processed
         bool is_favorite = false;
+        size_t generation = 0;
```

Comment:
Every item in the game gets size_t larger so you can handle a sorting issue in a menu? No thanks, there is definitely a local fix to this in inventory_column itself, we just need to find it.

Reply:
Can I convince you that the pointer comparison is fine? It's only used to disambiguate between identically named items and it doesn't need to be deterministic from the player's perspective. The sort order between such items is currently undefined so it wouldn't make things worse in any case.

Another option is to loop through contents like `item::stacks_with()`.

Reply:
I think I've found a solution with https://github.com/CleverRaven/Cataclysm-DDA/pull/53289/commits/9b5caf8afaf5e993a7fecfd42b6e793f2a397e67; please take a look. I've changed `inventory_selector` so that contents are always added right after their parent. In combination with `std::stable_sort` this should lead to the desired ordering.

However, the sort order between identical items can still change when they are dragged/moved/etc but at least this solution doesn't feel like a hack and doesn't require a lot of looping.